### PR TITLE
Add LogConstant time stepper

### LINKED ISSUE #9312: Add LogConstant time stepper
### Description of the enhancement or error report
Add a TimeStepper class that would set the time step so that it remains constant in the logarithm space.

### Rationale for the enhancement or information for reproducing the error
Logarithmic time steps are useful for the modeling of long-term behavior of materials or structures, notably in concrete creep or durability.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
New TimeStepper class to add, nothing else changed.

### PR DESCRIPTION
**Pull request by agiorla**
- Added a LogConstantDT time stepper and appropriate test

closes #9312

### CODE REVIEW

File: framework/src/timesteppers/LogConstantDT.C [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Real
+LogConstantDT::computeInitialDT()
+{
+  return _first_dt;
+}
+
+Real
+LogConstantDT::computeDT()
+{
+  Real next = std::pow(10., std::log10(_time) + _log_dt);
```
Comment (dschwen) [inline framework/src/timesteppers/LogConstantDT.C]: you can compute `_dt_factor = std::pow(10.0, _log_dt)` _once_ and then do

```
const Real next = _time * _dt_factor;
```

File: framework/src/timesteppers/LogConstantDT.C [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#include "LogConstantDT.h"
+
+template <>
+InputParameters
+validParams<LogConstantDT>()
+{
+  InputParameters params = validParams<TimeStepper>();
+  params.addClassDescription(
+      "TimeStepper which imposes a time step constant in the logarithmic space");
+  params.addRequiredParam<Real>("dt", "Time step in log10(time)");
```
Comment (dschwen) [inline framework/src/timesteppers/LogConstantDT.C]: calling this parameter dt is a bit confusing

File: test/tests/time_steppers/logconstant_dt/tests [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[Tests]
+  [./logconstant_dt]
+    type = 'Exodiff'
+    input = 'logconstant_dt.i'
+    exodiff = 'logconstant_dt_out.e'
```
Comment (dschwen) [inline test/tests/time_steppers/logconstant_dt/tests]: If you want to be thrifty then you make this a csv test. Afterall, all you need to test is that you are hitting the same times.

File: framework/src/timesteppers/LogConstantDT.C (hunk 2 above)
Comment (agiorla) [inline framework/src/timesteppers/LogConstantDT.C]: That's true, I will change that.

File: framework/src/timesteppers/LogConstantDT.C (hunk 1 above)
Comment (agiorla) [inline framework/src/timesteppers/LogConstantDT.C]: Yes, we can indeed avoid a few calculations here, thanks.

File: test/tests/time_steppers/logconstant_dt/tests (hunk 3 above)
Comment (agiorla) [inline test/tests/time_steppers/logconstant_dt/tests]: I used the same test than for the other time steppers actually, to remain consistent between the different tests. Should I still go for a CSV output?

Comment (dschwen) [inline test/tests/time_steppers/logconstant_dt/tests]: I don't feel strongly about this.

**Review by permcody — APPROVED** [review]

**Comment by permcody** [comment]
Thanks @agiorla for the contribution!

**Pull request merged**